src/System.Text.Json/src/System/Text/Json/Node/JsonObject.cs

```diff
@@ -0,0 +1,392 @@
+        /// </exception>
+        [CLSCompliant(false)]
+        public void Add(string propertyName, ulong propertyValue) => _dictionary.Add(propertyName, new JsonNumber(propertyValue));
+
+        /// <summary>
+        ///   Adds the specified property as a <see cref="JsonNumber"/> to the JSON object.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to add.</param>
+        /// <param name="propertyValue"><see cref="decimal"/> value of the property to add.</param>
+        /// <exception cref="ArgumentException">
+        ///   Property name to set already exists if handling duplicates is set to <see cref="DuplicatePropertyNameHandling.Error"/>.
+        /// </exception>
+        public void Add(string propertyName, decimal propertyValue) => _dictionary.Add(propertyName, new JsonNumber(propertyValue));
+
+        /// <summary>
+        ///   Adds the properties from the specified collection to the JSON object.
+        /// </summary>
+        /// <param name="jsonProperties">Properties to add.</param>
+        /// <exception cref="ArgumentException">
+        ///   Provided collection contains duplicates if handling duplicates is set to <see cref="DuplicatePropertyNameHandling.Error"/>.
+        /// </exception>
+        public void AddRange(IEnumerable<KeyValuePair<string, JsonNode>> jsonProperties)
+        {
+            foreach (KeyValuePair<string, JsonNode> property in jsonProperties)
+            {
+                Add(property);
+            }
+        }
+
+        /// <summary>
+        ///   Removes the property with the specified name.
+        /// </summary>
+        /// <param name="propertyName"></param>
+        /// <returns></returns>
+        public bool Remove(string propertyName) => _dictionary.Remove(propertyName);
+
+        /// <summary>
+        ///   Determines whether a property is in a JSON object.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to check.</param>
+        /// <returns>
+        ///   <see langword="true"/> if the property is found in a JSON object,
+        ///   <see langword="false"/> otherwise.
+        /// </returns>
+        public bool ContainsProperty(string propertyName) => _dictionary.ContainsKey(propertyName);
+
+        /// <summary>
+        ///   Modifies the name of specified property.
+        /// </summary>
+        /// <param name="oldName">Old name of the property to change.</param>
+        /// <param name="newName">New name of the property to change.</param>
+        /// <exception cref="ArgumentException">
+        ///   Property name to set already exists if handling duplicates is set to <see cref="DuplicatePropertyNameHandling.Error"/>.
+        /// </exception>
+        /// <exception cref="KeyNotFoundException">
+        ///   Property with specified name is not found in JSON object.
+        /// </exception>
```

Comment:
Worth adding a remark about ordering preservation (i.e. this makes no guarantees about keeping ordering the same)?